packages/web/src/routes/apps.tsx

```diff
@@ -55,20 +56,24 @@ export function AppsPage() {
   return (
     <AppProvider key={app.id} app={app}>
       <FilesProvider channel={channelRef.current} rootDirEntries={rootDirEntries}>
-        <PreviewProvider channel={channelRef.current}>
-          <LogsProvider channel={channelRef.current}>
-            <PackageJsonProvider channel={channelRef.current}>
+        <LogsProvider channel={channelRef.current}>
+          <PackageJsonProvider channel={channelRef.current}>
+            <PreviewProvider channel={channelRef.current}>
               <Apps />
-            </PackageJsonProvider>
-          </LogsProvider>
-        </PreviewProvider>
+            </PreviewProvider>
+          </PackageJsonProvider>
+        </LogsProvider>
       </FilesProvider>
     </AppProvider>
```

Comment:
NOTE: this context ordering is important! `PreviewProvider` uses `PackageJsonProvider`, and `PackageJsonProvider` uses `LogsProvider`.

Reply:
Good type of comment to add in code. Nice to find that if you're about to refactor.